Sources/HummingbirdFoundation/Files/FileMiddleware.swift

```diff
@@ -80,121 +84,140 @@ public struct HBFileMiddleware: HBMiddleware {
-                                // override content-length set above
-                                headers.replaceOrAdd(name: "content-length", value: String(describing: upperBound - lowerBound + 1))
-
-                                return HBResponse(status: .partialContent, headers: headers, body: body)
-                            }
+                        if let contentSize = contentSize {
+                            let lowerBound = max(range.lowerBound, 0)
+                            let upperBound = min(range.upperBound, contentSize - 1)
+                            headers.replaceOrAdd(name: "content-range", value: "bytes \(lowerBound)-\(upperBound)/\(contentSize)")
+                            // override content-length set above
+                            headers.replaceOrAdd(name: "content-length", value: String(describing: upperBound - lowerBound + 1))
+                        }
+                        return .loadFile(headers, range)
                     }
                 }
-                return self.fileIO.loadFile(path: fullPath.relativePath, context: request.context, logger: request.logger)
-                    .map { body in
-                        return HBResponse(status: .ok, headers: headers, body: body)
-                    }
+                return .loadFile(headers, nil)
+            }.flatMap { (result: FileResult) -> EventLoopFuture<HBResponse> in
+                switch result {
+                case .response(let response):
+                    return request.eventLoop.makeSucceededFuture(response)
+                case .loadFile(let headers, let range):
+                    switch request.method {
+                    case .GET:
+                        if let range = range {
+                            return self.fileIO.loadFile(path: fullPath.relativePath, range: range, context: request.context, logger: request.logger)
+                                .map { body, _ in
+                                    return HBResponse(status: .partialContent, headers: headers, body: body)
+                                }
+                        }
+                        return self.fileIO.loadFile(path: fullPath.relativePath, context: request.context, logger: request.logger)
+                            .map { body in
+                                return HBResponse(status: .ok, headers: headers, body: body)
+                            }
 
-            case .HEAD:
-                return request.success(HBResponse(status: .ok, headers: headers, body: .empty))
+                    case .HEAD:
+                        return request.success(HBResponse(status: .ok, headers: headers, body: .empty))
 
-            default:
-                return request.failure(error)
+                    default:
+                        return request.failure(error)
+                    }
+                }
             }
         }
     }
+
+    private enum FileResult {
+        case response(HBResponse)
```

Comment:
`response` really is a `notModified` response, right?

Reply:
If so, I'd rename it to be more clear

Reply:
Yeah I guess, left it to be open to change but it is internal so can change at a later date if needed

Reply:
done